src/components/ui/Input.tsx

```diff
@@ -1,34 +1,46 @@
-import { ChangeEvent, useState } from "react";
+import { ChangeEvent, useState } from "react"
 import { AiFillEye, AiFillEyeInvisible } from "react-icons/ai"
 
 interface InputProps extends React.HTMLAttributes<HTMLInputElement> {
-  type?: string
-  value?: string;
-  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
+  type: string
```

Comment:
I made it mandatory because I noticed at times there was type text being used when it should be a different type such as number. Reply to this to let me know if you want it to be optional but i wouldn't recommend it because the code becomes more prone to error.